ci-configs/.gitlab-ci.yml

```diff
@@ -0,0 +1,67 @@
+image: node:8
+
+# Cache dependencies.
+cache:
+  paths:
+    - node_modules/
+
+# Define the desired stages.
+test:
+  stage: test
+  script:
+    - yarn install
+    - yarn test
+
+lint:
+  stage: test
+  script:
+    - yarn install
+    - yarn lint
+
+# Code quality setup.
+code_quality:
+  stage: test
+  image: docker:stable
+  variables:
+    DOCKER_DRIVER: overlay2
+  allow_failure: true
+  services:
+    - docker:stable-dind
+  script:
+    - export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
+    - docker run
+      --env SOURCE_CODE="$PWD"
+      --volume "$PWD":/code
+      --volume /var/run/docker.sock:/var/run/docker.sock
+      "registry.gitlab.com/gitlab-org/security-products/codequality:$SP_VERSION" /code
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+
+# Auto-deployment to GitLab pages.
+pages:
+  stage: deploy
+  script:
+    # https://docs.gitlab.com/ee/user/project/pages/getting_started_part_four.html#the-public-directory
+    # Move the production-ready version into the `public` folder.
+    - yarn run build
+    - mv generated_build_folder public
+    - cd public && touch .nojekyll
```

Comment:
Does this actually deploy or just prepares the deployment?

Reply:
It both prepares and deploys the project to GL pages - [**c-hive.dev**](https://gitlab.com/c-hive/c-hive.dev/blob/master/.gitlab-ci.yml) also uses the same format.